cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -271,7 +307,7 @@ def project_dual(self):
 
     projected_dual_var = {'lambda_pos': projected_lambda_pos,
                           'lambda_neg': projected_lambda_neg,
-                          'lambda_lu': projected_lambda_lu,
+                          'lambda_lu': new_projected_lambda_lu,
                           'lambda_quad': projected_lambda_quad,
                           'nu': projected_nu}
```

Comment:
I suspect that the recursive call to DualFormulation here might make it impossible to cache everything without a refactor. I was working through caching as many things as possible and seeing whether or not I run into a wall.